### DISCUSSION TITLE
Make client.close(); in the RemoteAdapter's sendRequest optional

### DISCUSSION THREAD

**Discussion by andreystavitsky** [flutterdata/flutter_data#228] [Ideas]
It can be useful if HTTP client is singleton and shared between all instances in the application. Also it is useful for HTTP2

**Comment by frank06** [comment]
Where would you close the client then? Issues can arise if you don't close the connection: https://pub.dev/documentation/http/latest/http/Client/close.html

```
It's important to close each client when it's done being used; failing to do so can cause the Dart process to hang.
```

**Comment by andreystavitsky** [comment]
https://github.com/dart-lang/http/issues/422

It depends

**Reply by frank06** [reply]
When should it be closed?

**Reply by andreystavitsky** [reply]
I think, if there's custom http.Client overridden, then it should be decided in the custom http.Client. Or by default it should be closed after every request, but ability to override that behaviour should be provided without overriding whole sendRequest method.

**Reply by frank06** [reply]
Like this? https://github.com/flutterdata/flutter_data/commit/a6f62677c6359898907dc17062372227095a6485

**Reply by andreystavitsky** [reply]
exactly, thank you very much!

**Reply by frank06** [reply]
It's released on `1.5.12`
